x-pack/plugins/security_solution/public/management/pages/endpoint_hosts/view/index.test.tsx

```diff
@@ -746,6 +774,122 @@ describe('when on the endpoint list page', () => {
       expect(renderResult.getByTestId('endpointDetailsActionsButton')).not.toBeNull();
     });
 
+    describe('when showing Activity Log panel', () => {
+      let renderResult: ReturnType<typeof render>;
+      const agentId = 'some_agent_id';
+
+      let getMockData: () => ActivityLog;
+      beforeEach(async () => {
+        global.IntersectionObserver = jest.fn(() => ({
+          root: null,
+          rootMargin: '',
+          thresholds: [],
+          takeRecords: jest.fn(),
+          observe: jest.fn(),
+          unobserve: jest.fn(),
+          disconnect: jest.fn(),
+        }));
+
+        const fleetActionGenerator = new FleetActionGenerator(Math.random().toString());
```

Comment:
Q. you probably should use a static seed to the generator in tests, so that the generated content is randomly always the same. perhaps:

```suggestion
        const fleetActionGenerator = new FleetActionGenerator('seed');
```

Reply:
done 7e962e4f300012c6d0b84d6b88bc241177478700